app/components/Home.tsx

```diff
@@ -0,0 +1,368 @@
+        ComponentDataKey.Mode
+      );
+      if (savedMode) {
+        this.setModeFromModeType(savedMode);
+      }
+      this.setState({
+        platform: this.editorKit.internal.componentManager.platform,
+      });
+    } catch {}
+  };
+
+  setModeFromModeType = (value: ModeType) => {
+    // Can be refactored to use find
+    // const mode = modes.find // where mode.type = value
+    for (const mode of modes) {
+      if (mode.type === value) {
+        this.logDebugMessage('setModeFromModeType mode: ', mode.type);
+        this.setState(
+          {
+            mode,
+          },
+          () => {
+            this.updatePreviewText();
+          }
+        );
+        return;
+      }
+    }
+  };
+
+  changeMode = (mode: Mode) => {
+    this.setState(
+      {
+        mode,
+      },
+      () => {
+        this.updatePreviewText();
+      }
+    );
+    this.logDebugMessage('changeMode mode: ', mode.type);
+    this.editorKit.internal.componentManager.setComponentDataValueForKey(
+      ComponentDataKey.Mode,
+      mode.type
+    );
+  };
+
+  configureMarkdown = () => {
+    // Remember old renderer, if overriden, or proxy to default renderer
+    const defaultRender =
+      MarkdownParser.renderer.rules.link_open ||
+      function (tokens: any, idx: any, options: any, env: any, self: any) {
+        return self.renderToken(tokens, idx, options);
+      };
+
+    MarkdownParser.renderer.rules.link_open = function (
+      tokens: any,
+      idx: any,
+      options: any,
+      env: any,
+      self: any
+    ) {
+      // If you are sure other plugins can't add `target` - drop check below
+      let aIndex = tokens[idx].attrIndex('target');
+
+      if (aIndex < 0) {
+        tokens[idx].attrPush(['target', '_blank']); // add new attribute
+      } else {
+        tokens[idx].attrs[aIndex][1] = '_blank'; // replace value of existing attr
+      }
+
+      // pass token to default renderer.
+      return defaultRender(tokens, idx, options, env, self);
+    };
+  };
+
+  updatePreviewText = () => {
+    const preview = document.getElementById(HtmlElementId.Preview);
+    if (preview) {
+      preview.innerHTML = MarkdownParser.render(this.state.text);
+    }
+  };
+
+  removeSelection = () => {
+    if (window.getSelection) {
+      window.getSelection().removeAllRanges();
+    }
+  };
+
+  configureResizer = () => {
+    const simpleMarkdown = document.getElementById(
+      HtmlElementId.SimpleMarkdown
+    );
+    const editor = document.getElementById(HtmlElementId.Editor);
+    const columnResizer = document.getElementById(HtmlElementId.ColumnResizer);
+    let pressed = false;
+    let startWidth = editor.offsetWidth;
+    let startX;
+    let lastDownX;
+
+    let resizerWidth = columnResizer.offsetWidth;
+    let safetyOffset = 15;
+
+    columnResizer.addEventListener(MouseEvent.Down, (event) => {
+      pressed = true;
+      lastDownX = event.clientX;
+      columnResizer.classList.add('dragging');
```

Comment:
These CSS classes should be constantized into an enum for safety, since they are used in multiple places.

Reply:
Done in https://github.com/sn-extensions/markdown-basic/pull/29/commits/3c26a0887e7d4a1e3ab44eaf0784ccf7bfd1d665